hudi-common/src/main/java/org/apache/hudi/common/table/timeline/TimelineDiffHelper.java

```diff
@@ -57,35 +57,38 @@ public static TimelineDiffResult getNewInstantsForIncrementalSync(HoodieTimeline
       List<HoodieInstant> newInstants = new ArrayList<>();
 
       // Check If any pending compaction is lost. If so, do not allow incremental timeline sync
-      List<Pair<HoodieInstant, HoodieInstant>> compactionInstants = getPendingCompactionTransitions(oldT, newT);
-      List<HoodieInstant> lostPendingCompactions = compactionInstants.stream()
+      List<Pair<HoodieInstant, HoodieInstant>> viewChangingInstants = getPendingActionTransitions(oldT, newT);
+      List<HoodieInstant> lostPendingActions = viewChangingInstants.stream()
           .filter(instantPair -> instantPair.getValue() == null).map(Pair::getKey).collect(Collectors.toList());
-      if (!lostPendingCompactions.isEmpty()) {
-        // If a compaction is unscheduled, fall back to complete refresh of fs view since some log files could have been
+      if (!lostPendingActions.isEmpty()) {
+        // If a compaction/clustering is unscheduled, fall back to complete refresh of fs view since some log files could have been
         // moved. Its unsafe to incrementally sync in that case.
-        LOG.warn("Some pending compactions are no longer in new timeline (unscheduled ?). They are :"
-            + lostPendingCompactions);
+        LOG.warn("Some pending view changing instants are no longer in new timeline (unscheduled ?). They are :"
+            + lostPendingActions);
         return TimelineDiffResult.UNSAFE_SYNC_RESULT;
       }
-      List<HoodieInstant> finishedCompactionInstants = compactionInstants.stream()
-          .filter(instantPair -> instantPair.getValue().getAction().equals(HoodieTimeline.COMMIT_ACTION)
-              && instantPair.getValue().isCompleted())
+      List<HoodieInstant> finishedViewChangingInstants = viewChangingInstants.stream()
```

Comment:
Can you construct a timeline and call timeline.viewAlteringInstants() instead to avoid duplicating the logic ?